bitcoin/src/blockdata/script/mod.rs

```diff
@@ -170,20 +170,19 @@ pub fn write_scriptint(out: &mut [u8; 8], n: i64) -> usize {
 ///
 /// This code is based on the `CScriptNum` constructor in Bitcoin Core (see `script.h`).
 pub fn read_scriptint(v: &[u8]) -> Result<i64, Error> {
-    let len = v.len();
-    if len > 4 {
-        return Err(Error::NumericOverflow);
-    }
     let last = match v.last() {
         Some(last) => last,
         None => return Ok(0),
     };
+    if v.len() > 4 {
+        return Err(Error::NumericOverflow);
+    }
     // Comment and code copied from Bitcoin Core:
     // https://github.com/bitcoin/bitcoin/blob/447f50e4aed9a8b1d80e1891cda85801aeb80b4e/src/script/script.h#L247-L262
     // If the most-significant-byte - excluding the sign bit - is zero
     // then we're not minimal. Note how this test also rejects the
     // negative-zero encoding, 0x80.
-    if (last & 0x7f) == 0 {
+    if (*last & 0x7f) == 0 {
```

Comment:
Curly change! If it is warranted I think it should really have had its own patch if not PR. 

FTR I had to write some code to convince myself the original was even right.
```rust
fn main() {
    let x = 0x80;
    let p = &x;

    println!("{:p}", p);
    if p & 0x80 > 0 {
        println!("Deref must have been called");
    } else {
        println!("Deref not called");
    }
}
```
Since I did all that to educate myself I'm going to ack this.

Reply:
Hey you never know when suddenly bitwise and on a `&u8` and `u8` might cause UB..... /s

I can revert it if anyone isn't comfortable.